### PR TITLE
Fix site and channel api bug

### LINKED ISSUE #98: Unable to receive on multiple channels
While attempting to set up a gateway listening on a wildcard address I found that the gateway currently doesn't listen on multiple addresses. Instead, the last configured receive channel is used. 

If you configure a wildcard channel last, then you get an error (because you can't use a wildcard receive channel as the default channel).

### LINKED ISSUE #101: Sending to multiple sites doesn't work
### Symptoms

When sending a message to multiple sites, only the last registered site will get the message.

### Who's affected

v3.0.0 user sending to more than one site.

### LINKED ISSUE #103: Only the last registered receive channel is used
### Symptoms

When registering more than one receive channel, only the last registered channel is used.

### Who's affected

v3.0.0 Users configuring more than one receive channel

### Details

Originally raised by @welshdave  in https://github.com/Particular/NServiceBus.Gateway/issues/98

### PR DESCRIPTION
**Pull request by andreasohlund**
Fixes #103
Fixes #101

### CODE REVIEW

**Comment by andreasohlund** [comment]
@bording @timbussmann ready for a final review

File: src/NServiceBus.Gateway.AcceptanceTests/When_listening_on_wildcard_uri.cs [hunk 1]
```diff
@@ -25,7 +25,30 @@ public EndpointWithWildCardUriAsDefault()
...
+                });
+            }
+        }
+
+        [Test]
+        public void Should_not_throw_exception_if_wildcard_channel_is_not_default()
+        {
+            Assert.DoesNotThrowAsync(async () =>
+                await Scenario.Define<ScenarioContext>()
+                    .WithEndpoint<EndpointWithWildCardUriAndFullyQualifiedDefault>(e => e.When(b => Task.FromResult(0)))
```
Comment (timbussmann) [inline src/NServiceBus.Gateway.AcceptanceTests/When_listening_on_wildcard_uri.cs]: `e => e.When(b => Task.FromResult(0))` what's this?
if you want to make sure the endpoint starts without an exception, use `.Done(c => c.EndpointsStarted)` instead.

File: src/NServiceBus.Gateway.AcceptanceTests/When_sending_a_message_to_multiple_sites.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    using AcceptanceTesting;
+    using NUnit.Framework;
+
+    public class When_sending_a_message_to_multiple_sites : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_be_able_to_reply_to_messages_from_all_sites()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Headquarters>(b => b.When(async (bus, c) => await bus.SendToSites(new[]
```
Comment (timbussmann) [inline src/NServiceBus.Gateway.AcceptanceTests/When_sending_a_message_to_multiple_sites.cs]: tbh I find it a bit confusing that a send operation can send to multiple destinations. Normal sends cannot do that.

File: src/NServiceBus.Gateway.AcceptanceTests/When_listening_on_multiple_uris.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                }
+                catch (WebException)
+                {
+                }
+            }
+        }
+
+        static HttpWebRequest CreateWebRequest(string uri)
+        {
+#pragma warning disable DE0003 // API is deprecated
```
Comment (timbussmann) [inline src/NServiceBus.Gateway.AcceptanceTests/When_listening_on_multiple_uris.cs]: is this API really deprecated? removing the comment locally doesn't seem to cause any issues.

**Comment by andreasohlund** [comment]
Ugh, saw this just as I merged, will deal with this on the next PR instead

**Pull request merged**